KIPs/kip-163.md

```diff
@@ -0,0 +1,571 @@
+
+The function emits `RequestCancelWithdrawal` event.
+
+```solidity
+function claim(uint256 _requestId);
+```
+
+Calls `withdrawApprovedStaking` to CNv3 internally. If the withdrawal was canceled, it must revive user’s shares accordingly. Note that KLAY in withdrawal requests didn’t receive any rewards.
+
+|               | User gives | User receives                                                          |
+| ------------- | ---------- | ---------------------------------------------------------------------- |
+| Claim success | -          | `KLAY` in withdrawal request                                           |
+| Claim failure | -          | `previewDeposit(KLAY in withdrawal request)`: Floored amount of shares |
+
+The function validates the following requirement(s):
+
+- The function MUST revert if a user attempts to claim another user’s request.
+- The function MUST revert if `withdrawApprovedStaking` call reverts.
+
+The function emits `Claimed` if withdrew successfully.
+
+```solidity
+function redelegateByAssets(address _targetCnV3, uint256 _assets);
+function redelegateByShares(address _targetCnV3, uint256 _shares);
+```
+
+Call `redelegate` to CNv3 internally. It must burn corresponding shares of users. User will use `byAssets` to redelegate the `exact amount of KLAY`, otherwise use `byShares` to redelegate the `exact amount of shares`. Note that user must use `byShares` to redelegate all the KLAY same reason as `redeem`. User must receive the corresponding shares by PD connected to targetCNv3. In below tables, $\color{blue}{blue}$ is for target, while $\color{red}{red}$ is from.
+
+|                                      | User gives                                                                       | User receives                                                                                                        |
+| ------------------------------------ | -------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------- |
+| `redelegateByAssets(target, KLAY)`   | $\color{red}{previewWithdraw(KLAY)}$: Ceiled amount of shares to redelegate KLAY | $\color{blue}{previewDeposit(KLAY)}$: Floored amount of shares                                                       |
+| `redelegateByShares(target, shares)` | `shares`                                                                         | $\color{blue}{previewDeposit(KLAY)}$: Floored amount of shares, where `KLAY` is $\color{red}{previewRedeem(shares)}$ |
+
+The function validates the following requirement(s):
+
+- The function MUST revert if `_targetCnV3` isn’t a valid CNv3 staking contract.
+- The function MUST revert if current CNv3 doesn’t allow redelegation.
+- The function MUST revert if a user redelegates more KLAY than staked.
+- The function MUST revert is a user redelegates 0 KLAY.
```

Comment:
typo
"is" -> "if"

Reply:
Applied in [376d6b3](https://github.com/klaytn/kips/pull/164/commits/376d6b367890bff5833efc94b7a838225587c1b1)